sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -248,10 +249,15 @@ case class PreprocessTableCreation(catalog: SessionCatalog) extends Rule[Logical
         DDLUtils.checkTableColumns(tableDesc.copy(schema = analyzedQuery.schema))
 
         val output = analyzedQuery.output
-        val partitionAttrs = normalizedTable.partitionColumnNames.map { partCol =>
-          output.find(_.name == partCol).get
```

Comment:
@HyukjinKwon do you know if the original code is correct? The attribute references here are compared by their names and not ids

Reply:
Ok nvm, should be fine, since `partitionColumnNames` are just strings